lib/matplotlib/gridspec.py

```diff
@@ -211,12 +223,41 @@ def __init__(self, nrows, ncols,
         self.wspace = wspace
         self.hspace = hspace
 
+        self.figure = fig
+
         GridSpecBase.__init__(self, nrows, ncols,
                               width_ratios=width_ratios,
                               height_ratios=height_ratios)
 
+        if (self.figure is None) or not(self.figure.get_constrained_layout()):
+            # warnings.warn("GridSpec must be called with the fig keyword "
```

Comment:
decide whether to put a warning or not :p

Reply:
I decided to `_log.info` it, because we can do that now!